### PR TITLE
add deps to shell, add API_TOKEN_FILE feature

### PR DESCRIPTION
**Pull request by ryantm**

### CODE REVIEW

**Comment by ryantm** [comment]
I am now using this in nixpkgs-update production, yay!

**Comment by Synthetica9** [comment]
What's the purpose behind the -p flag? That should already get loaded from
shell.nix, and I'd like to keep it centralised if possible

**Comment by ryantm** [comment]
It might be because I'm ignorant of how to use the default.nix here. How would you do an "import from derivation" on this repo? By using the -p in the shebang, I didn't need to use shell.nix or default.nix at all, however, this means it was building the dependencies at runtime, so probably figuring out the import from derivation way would be better.

**Comment by ryantm** [comment]
For reference, here is how I'm using this https://github.com/nix-community/infra/blob/9f50926734dfb1079425a15ef4576e0859cbcad9/build01/nixpkgs-update.nix#L19

**Comment by Synthetica9** [comment]
Oh, I see. 

By default, nix-shell without arguments relies on loading `shell.nix` from our working directory. However, paths are still treated absolutely, so we can say `#!nix-shell -i python3 ./shell.nix` instead and have it work for any working directory. Does that work for you?

File: main.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env nix-shell
-#!nix-shell -i python3
+#! nix-shell -i python3 -p "python3.withPackages(ps: with ps; [requests dateutil libversion cachecontrol lockfile])"
```
Comment (Synthetica9) [inline main.py:2]:
```suggestion
#! nix-shell -i python3 ./shell.nix
```

**Comment by Synthetica9** [comment]
Closed via 222a5199778c38151a079989900a117853eea6b1 and 976854559e4ecde96c43de40dd90b7b5e5f9966a

**Pull request closed without merging**